# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.h
# Commit: 08eb9cf235bb
# Full Hash: 08eb9cf235bb5334ef01075d6148e1ea235adf56
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-05 04:07:37
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r ff3135366145 -r 08eb9cf235bb dom/ipc/WindowGlobalParent.h
--- a/dom/ipc/WindowGlobalParent.h	Wed Jun 04 19:25:54 2025 +0000
+++ b/dom/ipc/WindowGlobalParent.h	Wed Jun 04 19:25:54 2025 +0000
@@ -318,19 +318,6 @@
 
   mozilla::ipc::IPCResult RecvReloadWithHttpsOnlyException();
 
-  mozilla::ipc::IPCResult RecvGetIdentityCredential(
-      IdentityCredentialRequestOptions&& aOptions,
-      const CredentialMediationRequirement& aMediationRequirement,
-      bool aHasUserActivation, const GetIdentityCredentialResolver& aResolver);
-  mozilla::ipc::IPCResult RecvDisconnectIdentityCredential(
-      const IdentityCredentialDisconnectOptions& aOptions,
-      const DisconnectIdentityCredentialResolver& aResolver);
-  mozilla::ipc::IPCResult RecvSetLoginStatus(
-      LoginStatus aStatus, const SetLoginStatusResolver& aResolver);
-
-  mozilla::ipc::IPCResult RecvPreventSilentAccess(
-      const PreventSilentAccessResolver& aResolver);
-
   mozilla::ipc::IPCResult RecvGetStorageAccessPermission(
       bool aIncludeIdentityCredential,
       GetStorageAccessPermissionResolver&& aResolve);
@@ -347,6 +334,8 @@
   already_AddRefed<dom::PWebAuthnTransactionParent>
   AllocPWebAuthnTransactionParent();
 
+  already_AddRefed<dom::PWebIdentityParent> AllocPWebIdentityParent();
+
  private:
   WindowGlobalParent(CanonicalBrowsingContext* aBrowsingContext,
                      uint64_t aInnerWindowId, uint64_t aOuterWindowId,